go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/NVIDIA/nvidia-container-toolkit
 
-go 1.20
+go 1.22.2
```

Comment:
Should I revert this change @elezar ?

Reply:
We already specify `GOLANG_VERSION ?= 1.22.2` in `versions.mk`. We also don't actually depend on any go1.22 features.

I think something like:
```
go 1.21
toolchain 1.22.2
```
might be more applicable here.

Reply:
sure